test/functional/tag_controller_test.rb

```diff
@@ -148,9 +148,12 @@ def setup
     json = ActiveSupport::JSON.decode(@response.body)
 
     assert_not_nil json
-    assert :preview
-    assert :image
-      assert :tags
+    assert_equal tags(:spectrometer).notes.first.body_preview,
+    json.first[:preview]
+    assert_equal tags(:spectrometer).notes.first.main_image_url,
+    json.first[:image] 
+    assert_equal tags(:spectrometer).notes.first.tags,
```

Comment:
`tags(:spectrometer).notes.first.tags` will return an array of tags objects -- so we need to do `tags(:spectrometer).notes.first.tags.collect(&:name)` and it'll collect the tags' names. See if that works. 

Also - i noticed the tests ran in 1m 30 seconds, which means they stalled -- this happens sometimes -- they should take about 7-8 minutes. To restart them, you can close and reopen the pull request and they'll trigger again.